### PR TITLE
basic functionality

### PR DESCRIPTION
**Pull request by miherlosev**

### CODE REVIEW

**Comment by miherlosev** [comment]
I am waiting for https://github.com/DevExpress/testcafe/issues/1212

**Comment by miherlosev** [comment]
We decide to release a first version and show this to vue's community.
So, review please.
/cc  @DevExpress/testcafe-docs

**Comment by miherlosev** [comment]
ping @churkin @AlexanderMoskovkin @VasilyStrelyaev

File: test/index.js [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+        }
+    }
+});
+
+test('there is no Vue on the tested page', async t => {
+    await ClientFunction(() => window.Vue = null)();
+
+    const body = await VueSelector('body');
+
+    console.log(JSON.stringify(body));
```
Comment (churkin) [inline test/index.js]: Is it real needed?

File: test/index.js [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+        .expect(listVue.props.id).eql('list1')
+        .expect(listVue.computed.reversedId).eql('1tsil');
+});
+
+test('composite selector', async t => {
+    const listItemVue6 = await VueSelector('list list-item').nth(5).vue;
+
+    await t.expect(VueSelector('list list-item').count).eql(6)
+        .expect(listItemVue6.props.id).eql('list2-item3');
+});
```
Comment (AlexanderMoskovkin) [inline test/index.js]: Maybe it's better to change order of operations?

I think, it's better to check `list-item` elements count at first, and then check the last element.


```js
    const listItem = VueSelector('list list-item');

    await t.expect(listItem.count).eql(6);

    const listItemVue6 = await listItem.nth(5).vue;

    await t.expect(listItemVue6.props.id).eql('list2-item3');
```

File: test/index.js (hunk 1 above)
Comment (miherlosev) [inline test/index.js]: Yes, It' s my mistake

**Comment by miherlosev** [comment]
FPR

**Comment by miherlosev** [comment]
FPR

File: src/index.js [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+            }
+
+            function getState (instance) {
+                const props   = instance._props || instance.$options.props;
+                const getters = instance.$options.vuex && instance.$options.vuex.getters;
+                const result  = {};
+
+                Object.keys(instance._data)
+                    .filter(key => !(props && key in props) && !(getters && key in getters))
+                    .map(key => {
```
Comment (churkin) [inline src/index.js]: `map` --> `forEach` ?

File: src/index.js [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+                    else
+                        break;
+                }
+            }
+        }
+
+        function findFirstRootInstance () {
+            let instance = null;
+
+            walkDomNodes(document, (node) => {
```
Comment (churkin) [inline src/index.js]: `node =>`

File: src/index.js [hunk 5]
```diff
@@ -0,0 +1,150 @@
...
+
+            if (vueVersion < SUPPORTED_VUE_VERSION)
+                throw new Error('testcafe-vue-selectors supports Vue version 2.x and newer');
+        }
+
+        function walkDomNodes (node, fn) {
+            if (node.childNodes) {
+                for (let i = 0; i < node.childNodes.length; i++) {
+                    const childNode = node.childNodes[i];
+
```
Comment (churkin) [inline src/index.js]: extra new line

File: src/index.js [hunk 6]
```diff
@@ -0,0 +1,150 @@
...
+                }
+
+                for (let i = 0; i < node.$children.length; i++) {
+                    const childNode = node.$children[i];
+
+                    walkVueComponentNodes(childNode, tagIndex, checkFn);
+                }
+            }
+
+            walkVueComponentNodes(root, 0, (node, tagIndex) => {
```
Comment (churkin) [inline src/index.js]: `(node, tagIndex) => tags[tagIndex] === getComponentTag(node)`

**Comment by miherlosev** [comment]
FPR

File: src/index.js [hunk 7]
```diff
@@ -0,0 +1,145 @@
+import { Selector } from 'testcafe';
+
+export default (selector) => {
+    if (selector !== void 0 && typeof selector !== 'string')
+        throw new Error(`If selector option is present it should be a string, but it was ${typeof selector}`);
```
Comment (VasilyStrelyaev) [inline src/index.js]:
```
        throw new Error(`If the selector parameter is passed it should be a string, but it was ${typeof selector}`);
```

**Comment by miherlosev** [comment]
FPR

**Comment by miherlosev** [comment]
I've added the README.md page. @VasilyStrelyaev review this

**Pull request merged**